File: LeetCode Problems/101 to 200/137. Single Number II/137. Single Number II.java
class Solution {
    public int singleNumber(int[] nums) {
        Map<Integer, Integer> map = new HashMap<>();
        for (int ele : nums) {
            map.merge(ele, 1, Integer::sum);
        }
        int ans = -1;
        for (Map.Entry<Integer, Integer> entry : map.entrySet()) {
            if (entry.getValue() == 1) ans = entry.getKey();
        }
        return ans;
    }
}

File: LeetCode Problems/101 to 200/137. Single Number II/137. Single Number II.cpp
class Solution {
public:
    int singleNumber(vector<int> &nums) {
        map<int, int> mp;
        for (auto ele: nums) {
            mp[ele]++;
        }
        for (auto it: mp) {
            if (it.second == 1)
                return it.first;
        }
        return 0;
    }
};